src/main/java/com/minecolonies/coremod/client/gui/WindowListRecipes.java

```diff
@@ -111,7 +119,23 @@ public void updateElement(final int index, @NotNull final Pane rowPane)
             {
                 @NotNull final IRecipeStorage recipe = recipes.get(index);
                 final ItemIcon icon = rowPane.findPaneOfTypeByID(OUTPUT_ICON, ItemIcon.class);
-                icon.setItem(recipe.getPrimaryOutput());
+                if(recipe.getRecipeType() == RecipeStorageType.MULTI_OUTPUT)
```

Comment:
This comes with the teritory of writing Enums in the API.
Refactor this so that the recipe type can handle it itself.

Reply:
See switch to registry